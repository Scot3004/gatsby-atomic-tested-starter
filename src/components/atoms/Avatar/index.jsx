import styled from 'styled-components';

export const Avatar = styled.img`
  flex: 0 0 ${props => props.size};
  width: ${props => props.size};
  height: ${props => props.size};
  padding: 0.6rem;
`;

export default Avatar;
