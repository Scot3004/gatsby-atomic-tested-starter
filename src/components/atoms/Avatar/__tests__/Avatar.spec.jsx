import React from 'react';
import 'jest-styled-components';
import renderer from 'react-test-renderer';
import Avatar from '..';

describe('When an avatar is displayed', () => {
  const avatarSrc = 'src_stub';

  test('Then it will be rendered in the given size', () => {
    const tree = renderer
      .create(<Avatar size="64px" src={avatarSrc} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
