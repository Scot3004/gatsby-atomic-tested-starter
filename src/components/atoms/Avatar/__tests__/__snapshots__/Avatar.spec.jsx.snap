// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`When an avatar is displayed Then it will be rendered in the given size 1`] = `
.c0 {
  -webkit-flex: 0 0 64px;
  -ms-flex: 0 0 64px;
  flex: 0 0 64px;
  width: 64px;
  height: 64px;
  padding: 1rem;
}

<img
  className="c0"
  size="64px"
  src="src_stub"
/>
`;
